apps/teamsjs-diagnostic-app/src/components/sample/CustomScenario.tsx

```diff
@@ -2,24 +2,55 @@ import React, { useState } from 'react';
 import './CustomScenario.css';
 import { useDrop } from 'react-dnd';
 import apiComponents, { ApiComponent } from './ApiComponents';
-import AppInstallDialogAPIs from '../../apis/AppInstallDialogApi';
-import BarCodeAPIs from '../../apis/BarCodeApi';
+import { handleRunScenario } from '../../utils/HandleRunScenario';
+import { app } from '@microsoft/teams-js';
+
+app.initialize();
 
 const CustomScenario: React.FC = () => {
   const [searchQuery, setSearchQuery] = useState<string>('');
   const [customScenario, setCustomScenario] = useState<Array<{ api: ApiComponent, func: string, input?: string }>>([]);
+  const [showAddScenario, setShowAddScenario] = useState<boolean>(false);
+  const [newScenarioName, setNewScenarioName] = useState<string>('');
+  const [selectedApis, setSelectedApis] = useState<{ [key: string]: boolean }>({});
+  const [selectedFunctions, setSelectedFunctions] = useState<{ [key: string]: string }>({});
+  const [apiInputs, setApiInputs] = useState<{ [key: string]: string }>({});
+  const [savedScenarios, setSavedScenarios] = useState<Array<{ name: string, scenario: Array<{ api: ApiComponent, func: string, input?: string }> }>>([]);
+  const [showScenarioList, setShowScenarioList] = useState<boolean>(false);
 
-  const handleRunScenario = async () => {
+  const handleRunScenarioClick = async () => {
     console.log('Running custom scenario...');
+  
+    let isSuccess = true;
+  
     for (const { api, func, input } of customScenario) {
-      console.log(`Executing ${func} for ${api.title} with input: ${input}`);
-      // Execute the API function based on the selected function and input
+  
+      try {
+        console.log("Enter");
+        await handleRunScenario(api, func, input);
+        console.log(`Success: ${func} for ${api.title}`);
+        console.log("Exit");
+      } catch (error: any) {
+        console.log(`Failure: ${func} for ${api.title} - ${error.message}`);
+        console.log("Exit");
+        isSuccess = false;
+        console.log('Custom scenario failed.');
+        break; // Stop further execution if any API fails
+      }
+    }
+  
+    if (isSuccess) {
+      console.log('Custom scenario completed successfully.');
     }
   };
 
   const addToScenario = (api: ApiComponent, func: string, input?: string) => {
     console.log(`Adding ${func} for ${api.title} with input: ${input}`);
-    setCustomScenario([...customScenario, { api, func, input }]);
+    if (customScenario.length < 5) {
+      setCustomScenario([...customScenario, { api, func, input }]);
+    } else {
+      console.log('Maximum limit reached. Cannot add more APIs to the scenario.');
```

Comment:
It would be nice if this log statement said what the maximum scenario length is, just to reinforce the limitation with the user.